src/helper/convertColumnElementsToData.js

```diff
@@ -99,4 +112,85 @@ function convertColumnElementsToData(childComponents) {
   };
 };
 
-export default convertColumnElementsToData;
+/**
+ *
+ */
+function collectPropsAndTemplatesFromColumnData(column, columnProps, elementTemplates) {
+  columnProps.push(_extractPropsFromData(column));
+  _extractTemplatesFromData(elementTemplates, column);
+}
+
+/**
+ * Uses columnGetter to fetch column props and cell rendering templates
+ *
+ * @param {!function(number)} columnGetter
+ * @param {!number} columnsCount
+ * @return {{
+ *   columnGroupProps: !Array.<columnDefinition>,
+ *   columnProps: !Array.<columnDefinition>,
+ *   elementTemplates: !Object.<string, Array>,
+ *   useGroupHeader: boolean,
+ * }}
+ */
+function fetchColumnData(columnGetter, columnsCount) {
+  let columnProps = [];
+  let columnGroupProps = [];
+  let columnsWithDefinedGroupIndex = 0;
+  let columnGroups = [];
+
+  const elementTemplates = getDefaultElementTemplates();
+
+  // fetch column props and templates for each column
+  for (let index = 0; index < columnsCount; index++) {
+    collectPropsAndTemplatesFromColumnData(columnGetter({ index }), columnProps, elementTemplates);
+  }
+
+  // keep a map of group indexes which will be used to fetch the column groups
+  forEach(columnProps, (column) => {
+    const groupIndex = column.groupIdx;
+    if (groupIndex !== undefined && groupIndex !== null) {
+      columnGroups[groupIndex] = true;
+      columnsWithDefinedGroupIndex++;
+    }
+  });
+
+  // if group index was specified, it should be given for every column
+  if (columnsWithDefinedGroupIndex > 0) {
+    invariant(columnsWithDefinedGroupIndex === columnProps.length,
+      'group index if specified must be given for every column');
+  }
+
+  // fetch the column groups
+  for (let index = 0; index < Object.keys(columnGroups).length; index++) {
+    columnGroups[index] = Object.assign({}, columnGetter({ index, group: true }, { index }));
+  }
+
+  // the column groups can be specified out of order, but our reducers/selectors require
+  // it to be specified in order. So we'll sort it.
+  columnGroups.sort((a, b) => a.index - b.index);
```

Comment:
Basically, columns through the new API can put any index as the column group.
But FDT renders/calculates column groups in the order they're specified (even fixed/fixedRight are intra-ordered this way)

So we'll just sort it.